sirepo/package_data/static/js/sirepo.js

```diff
@@ -2147,4 +2225,47 @@ SIREPO.app.controller('SimulationsController', function (appState, panelState, r
         appState.saveQuietly('simulationFolder');
     });
     loadList();
+
+    // invoked in loadList() callback
+    function checkURLForFolder() {
+
+        var pathNoSectionRE = new RegExp('^\/' + activeSection.getActiveSection());
+        var decodedPath = decodeURI($location.path());
+        var canonicalPath = compoundPathToPath(decodedPath.replace(pathNoSectionRE, ''));
+
+        if( canonicalPath !== (appState.getActiveFolderPath() || '/') ) {
+            var fList = [rootFolder()];
+            var newFolder = folderForPathInList(canonicalPath, fList);
+            if( newFolder ) {
+                appState.setActiveFolderPath(canonicalPath);
+                appState.setActiveFolder(newFolder);
+                self.openItem(newFolder);
+            }
+            else {
+                requestSender.localRedirect('notFound');
+            }
+        }
+        else {
+            if( !appState.getActiveFolder() ) {
+                self.openItem(rootFolder());
+            }
+        }
+    }
+    function folderForPathInList (folderPath, folderList) {
+        for( var i = 0; i < folderList.length; ++i ) {
+            var item = folderList[i];
+            if( item.isFolder ) {
```

Comment:
short circuit

Reply:
`            if( !item.isFolder ) {
                continue;
            }
...
`
?

Reply:
Yes, but not one line:

```
if (! item.isFolder) {
    continue;
}
```

Note the spacing, btw. We put a space after the keyword and the paren, no spaces between the paren and the next token, space after `!` and the value it is operating on so it is more visible.